### ISSUE TITLE
Are the defaults supposed to be a 1:1 match for char-rnn?

### ISSUE THREAD

**Issue by clausd** [jcjohnson/torch-rnn#17]
I'm trying the shakespeare dataset on torch-rnn and char-rnn using all defaults in both cases. I'm training without a GPU. Before doing any kind of tweaking of model size and/or dropout it seems char-rnn gets a much better training loss and validation loss. Should the loss be approximately the same for these two models? torch-rnn is stuck at about 1.6 to 1.7 where char-rnn manages a considerably better loss with fewer epochs. 

A quick examination of sampled text seems to show that char-rnn has performed better as well: Almost every word sampled is an actual word, torch-rnn hasn't got the vocabulary right yet. 

Is this just a question of different defaults  for modelparameters/hyperparameters or is something wrong?

**Comment by jcjohnson** [comment]
It should not be exactly the same as char-rnn, because we use a word embedding layer while char-rnn uses one-hot encodings at the first layer. Also we optimize the model with Adam while char-rnn uses RMSProp. These shouldn't make a huge difference; but the default hyperparameters in torch-rnn might not be as good as those in char-rnn.

**Comment by clausd** [comment]
OK - I'm fairly consistently failing to get the same quality of training even though I'm experimenting with model size and dropout.

**Comment by AlekzNet** [comment]
I managed to get better results, than char-nn by removing resetstates (https://github.com/jcjohnson/torch-rnn/issues/46), using big seq_length (e.g. 10000), and aggressive learning decay  with high initial learning rate value (1e-2).

**Comment by ChrisCummins** [comment]
Marking this as solved. The model is different from char-rnn.
